### PR TITLE
Add target to approve new public API

### PR DESCRIPTION
**Pull request by blairconrad**
… I get tired of copying files manually.

### CODE REVIEW

**Comment by blairconrad** [comment]
Technically I think this target should be called "approve" and the current approve something like "check-api", but what can you do?
Also "check-api" is not the best name.

**Comment by thomaslevesque** [comment]
> Technically I think this target should be called "approve" and the current approve something like "check-api", but what can you do?
> Also "check-api" is not the best name.

I don't mind changing the name of the existing target to use `approve` for the new one. `check-api` isn't great, but I don't have a better idea...

File: tools/FakeItEasy.Build/Program.cs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 namespace FakeItEasy.Build
 {
+    using System;
```
Comment (thomaslevesque) [inline tools/FakeItEasy.Build/Program.cs]: What do you need this for? As far as I can tell, the new code only uses classes from `System.IO`

File: tools/FakeItEasy.Build/Program.cs [hunk 2]
```diff
@@ -63,6 +64,16 @@ public static void Main(string[] args)
...
 
+            Target(
+                "accept-api",
+                () =>
+                    {
+                        foreach (var received in Directory.EnumerateFiles("tests/FakeItEasy.Tests.Approval", "*.received.txt", SearchOption.AllDirectories))
+                        {
+                            File.Copy(received, received.Replace(".received.txt", ".approved.txt"), overwrite: true);
+                        }
+                    });
```
Comment (thomaslevesque) [inline tools/FakeItEasy.Build/Program.cs]: How do we use this? It assumes the .received files have already been generated... So, run `approve`, which will fail, then run `accept-api`?

File: tools/FakeItEasy.Build/Program.cs [hunk 3]
```diff
@@ -63,6 +64,16 @@ public static void Main(string[] args)
                 forEach: ProjectsToPack,
                 action: project => Run("dotnet", $"pack {project.Path} --configuration Release --no-build --output {Path.GetFullPath("artifacts/output")}"));
 
+            Target(
+                "accept-api",
+                () =>
+                    {
+                        foreach (var received in Directory.EnumerateFiles("tests/FakeItEasy.Tests.Approval", "*.received.txt", SearchOption.AllDirectories))
```
Comment (thomaslevesque) [inline tools/FakeItEasy.Build/Program.cs]: Maybe specify `/ApprovedApi` in the path, to further reduce the search scope ?

File: tools/FakeItEasy.Build/Program.cs (hunk 2 above)
Comment (thomaslevesque) [reply tools/FakeItEasy.Build/Program.cs]: Maybe we should add some kind of flag (e.g. an environment variable) to change the behavior of the approval tests. If the flag is set, don't compare received against approved, but just copy received to approved?

Comment (blairconrad) [reply tools/FakeItEasy.Build/Program.cs]: > So, run approve, which will fail, then run accept-api?

I usually build everything, find the failure on `approve` and then manually copy. I was just trying to ease the last step. Do you do something different?

> Maybe we should add some kind of flag (e.g. an environment variable) to change the behavior of the approval tests. If the flag is set, don't compare received against approved, but just copy received to approved?

Could do, I suppose. I guarantee you 100% I will forget.

File: tools/FakeItEasy.Build/Program.cs (hunk 1 above)
Comment (blairconrad) [reply tools/FakeItEasy.Build/Program.cs]: In initial development, I was just spewing messages to the console. I forgot to remove the `using` when I changed that.

**Comment by blairconrad** [comment]
> I don't mind changing the name of the existing target to use approve for the new one. check-api isn't great, but I don't have a better idea...

validate-api feels slightly better…

or we could leave as is. Or drop the whole thing if you like another approach, such as the environment variables.

File: tools/FakeItEasy.Build/Program.cs (hunk 2 above)
Comment (thomaslevesque) [reply tools/FakeItEasy.Build/Program.cs]: > I usually build everything, find the failure on `approve` and then manually copy. I was just trying to ease the last step. Do you do something different?

For me, each test that fail cause my diff tool (Devart Code Compare) to open the received and approved files, then I just need hit the "Copy all changes to right" button. Doesn't it do that for you?

> Could do, I suppose. I guarantee you 100% I will forget.

Yes, I'd probably forget too. Your way is fine, I think

**Comment by thomaslevesque** [comment]
> validate-api feels slightly better…
> 
> or we could leave as is. Or drop the whole thing if you like another approach, such as the environment variables.

Or keep the name of the current target, and name the new one `force-approve`?

File: tools/FakeItEasy.Build/Program.cs (hunk 2 above)
Comment (blairconrad) [reply tools/FakeItEasy.Build/Program.cs]: For me, it pops up copies of... something. WinMerge? I dunno. Then I panic and close them all and copy over! I could use the differ to copy.

**Comment by blairconrad** [comment]
I think I made your suggested changes. And I squashed. If you like it enough, we could merge. If you have any doubts, we could not!!

**Comment by thomaslevesque** [comment]
Thanks @blairconrad!

**Comment by blairconrad** [comment]
Thank you!

**Pull request merged**